CONTRIBUTING.md

```diff
@@ -1,5 +1,46 @@
+### Setup
+
+We recommend to create a virtual environment to contribute to the client. You can install all developer dependencies using `pip install requirements.txt`.
+
+### Testing
+
+We use [pytest](https://docs.pytest.org) to write tests, however there are plenty of older tests that use unittest.
+
+There are three kinds of tests:
+- Unit tests, that test individual components of the client
+- Integration tests, that test the client with a running weaviate instance
+- Mock tests, where a weaviate instance is mocked to return specific replies
+
+To run the integration tests start the weaviate instances using `./ci/start_weaviate.sh`.
+Then run all tests with
+```
+pytest integration
+pytest mock_tests
+pytest test
+```
+
+### Linting
+
 We use the following tools to ensure a high code quality:
 - black (formatter)
 - flake8 with plugins
```

Comment:
how to run these locally without configuring precommit hook? what Flake8 plugins are required?

Reply:
Added (but I'd really recomment using pre-commits)

> Flake8 plugins are required

They are part of the requirements.txt file, so no furhter action is needed here.